codegenerator/cli/templates/static/codegen/src/db/EntityHistory.res

```diff
@@ -217,6 +218,8 @@ let fromTable = (table: table, ~schema: S.t<'entity>): t<'entity> => {
 
   let actionField = mkField(actionFieldName, Custom(Enums.EntityHistoryRowAction.enum.name))
 
+  let serialField = mkField("serial", Serial, ~isNullable=true)
```

Comment:
> I think it should have an index

I agree, I want to add a few indexes actually.

> Or even a PRIMARY KEY

It could be primary key but we need a unique constraint on chainId, blockNumber, logIndex and id. Well it doesn't need to be enforced technically but it's already helped catch a few bugs. 

> Also, I don't understand why it's nullable, 

It's just because the serial is calculated on postgres side. Marking "nullable" here simply means the the field doesn't get a "NOT NULL" added on table creation. And you can pass "Null" on inserts to let postgres deal with the row creation.

Reply:
What about something like `NOT NULL DEFAULT nextval('table_name_id_seq')`?

Reply:
If it complicates things we can keep it as nullable